.github/workflows/deploy-integration-ec2.yml

```diff
@@ -0,0 +1,31 @@
+on:
+  push:
+    branches:
+      - master
+name: Deploy Integration
+jobs:
+  deploy-integration:
+    if: github.ref == 'refs/heads/master'
```

Comment:
If I'm understanding the logic here (which I could entirely not be), we're going to deploy integration on merges to master? But also, wouldn't merges to master deploy to production as well?

It would be great to deploy integration on merges to stage (or create a new branch, integration). And deploy to production on mergers to master.

Reply:
(completely tangential, we really should move off of master soon bc that terminology is horrible)

Reply:
Sure! so there's two logic blocks in each workflow file - lines 1-4 which trigger the action, and then the block here on line 7 which is defining the job `deploy-integration`. 
```
on:
  push:
    branches:
      - master
name: Deploy Integration
jobs:
  deploy-integration:
    if: github.ref == 'refs/heads/master'
```
the if statement here is admittedly a little redundant, but serves as another check to make sure that we run this action only if the [context](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context) in which this action was run is indeed the head of `master`. 
Practically, it's just another guard to confirm that we're only running our job in the contexts we deem appropriate. 


as far as the production deploy goes
```
on:
  push:
    tags:
      - 'v*'
name: Deploy Production
jobs:
  deploy-production-1:
    if: github.ref == 'refs/heads/master'
```
this is a good catch because this logic block will always eval to false - the `github.ref` should equal something in the format of `refs/tags/<tag number>`. The upside here is that we just wouldn't run the production release jobs - the action would eval the if as false and then skip running the job.

Reply:
> we're going to deploy integration on merges to master? But also, wouldn't merges to master deploy to production as well? It would be great to deploy integration on merges to stage (or create a new branch, integration). And deploy to production on mergers to master.

I think my intent is to try to mirror our current branching strategy -> deployment process as much as possible. I'm under the current impression that the workflow is to test `master` on integration and create release tags for production (and deploy those tags to production) if everything looks good in integration. If that's not the case then I'd want to revisit the triggers for these jobs - which wouldn't be hard to do at all.

Reply:
Let's revisit yes! I became the only engineer on the server back in October - so to create less PRs for non-autobuilder folks to review, I would just PR against master. Rather than bug them to review a PR against staging branch, test staging branch and then have to bug them against to review essentially same/similar PR against master.

However, the ideal/original workflow was to open a PR against staging and then test staging on integration environment. With multiple people working on the autobuilder, and thus more engineering resources for reviewing, I think it makes most sense to return to our original workflow.

Reply:
(also to make the terminology more clear, we could create a branch Integration and associate that with testing on the integration environment, rather than using stage as we did previously)

Reply:
That sound perfect - we're actually really close to https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow then. I'll make some changes to the triggers.

Reply:
@madelinezec I think I'll keep the trigger for production deploys as push of new release tags - this lets us keep the intentionality we have with manual deploys to production.